otx/hpo/hpo_runner.py

```diff
@@ -0,0 +1,387 @@
+"""HPO runner and resource manager class."""
+
+# Copyright (C) 2022 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+
+import logging
+import multiprocessing
+import os
+from abc import ABC, abstractmethod
+from copy import deepcopy
+from functools import partial
+from typing import Any, Callable, Dict, List, Literal, Optional, Union
+
+import torch
+
+from otx.hpo.hpo_base import HpoBase, Trial, TrialStatus
+from otx.hpo.utils import check_positive
+
+logger = logging.getLogger(__name__)
+
+
+class ResourceManager(ABC):
```

Comment:
This abstract class is awesome ! However, I think we need to make separate `.py` file for this if we consider the further implementation.

Reply:
I see. I'll reorganized classes as you said.